### DISCUSSION TITLE
package-lock.json generated using npm install giving loader errors

### DISCUSSION THREAD

**Discussion by Anushka-Hapukotuwa** [apache/superset#16202] [Q&A / Help]
When I clone superset, navigate to superset/superset-frontend/ and give npm ci and then npm run build, the project builds  and runs successfully.
Next I deleted the superset/superset-frontend/package-lock.json and generated it again from the superset/superset-frontend/package.json using npm install, then it gives loader errors.

Screenshots - 
![MicrosoftTeams-image (49)](https://user-images.githubusercontent.com/87460606/129025101-b5eae158-49ae-4c14-bb53-53767519dae3.png)
Error generated -
ERROR in ./src/datasource/ChangeDatasourceModal.tsx 112:118
Module parse failed: Unexpected token (112:118)
File was processed with these loaders:
 * ./node_modules/thread-loader/dist/cjs.js
 * ./node_modules/babel-loader/lib/index.js
 * ./node_modules/ts-loader/index.js
You may need an additional loader to handle the result of these loaders.
|   };
|   const changeSearch = (event) => {var _event$currentTarget;
>     const searchValue = ((_event$currentTarget = event.currentTarget) == null ? void 0 : _event$currentTarget.value) ?? '';
|     setFilter(searchValue);
|   };
 
Is there a problem with building the superset-frontend, when we delete the package-lock.json and regenerate it from package.json using npm install?

superset version - 1.2
npm version - 6.14
node version - 14.17.2

**Comment by foreground-voice** [comment]
I'm not supporter, but suggest the better way of building just make package.json with fixed versions and remove package-lock from repo.
